main.go

```diff
@@ -251,46 +389,48 @@ func main() {
 			logrus.WithField("zone", z.String()).Debug("Got mock zone seed")
 		}
 
-		dnsProvider = mockjson.NewProvider(zones, mockZoneStatePath)
+		dnsProvider = mockjson.NewProvider(zones, viper.GetString("mockZoneStatePath"))
 	}
 
-	if debug {
+	if viper.GetBool("debug") {
 		dnsProvider = NewDebugDNSProvider(dnsProvider)
 	}
 
 	// - Setup Controllers -----------------------------------------------------
 	metrics := &Metrics{}
-	err := metrics.Init()
-	if err != nil {
-		logrus.Fatalf("couldn't initialize metrics, see: %v", err)
+	metricsErr := metrics.Init()
+	if metricsErr != nil {
+		logrus.Fatalf("couldn't initialize metrics, see: %v", metricsErr)
 	}
 
 	loadbalancers := make([]Loadbalancer, 0)
-	for key, value := range lbs {
-		lb, err := parseLoadbalancer(key, value)
+	logrus.Warn("here")
+	viperLb := viper.GetStringMapStringSlice("lb")
+	logrus.Warn("here")
+	for key, value := range viperLb {
+		lb, err := parseLoadbalancerEndpointList(key, value)
 		if err != nil {
 			logrus.Fatalf("couldn't parse endpoints `%+v` for dns zone `%s`, see: %v", value, key, err)
 		}
-
 		loadbalancers = append(loadbalancers, lb)
 	}
 
 	lbUpdates := make(chan *LoadbalancerList, 0)
 
 	etcd := &ETCD{}
-	err = etcd.Init(etcds)
+	var err = etcd.Init(viper.GetStringSlice("etcEndpoint"))
```

Comment:
why not just err := etcd.Init.... ?

Reply:
Im still not golang idomatic, but fixed.